modules/product/src/ProductVariationManager.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\commerce_product\ProductVariationManager.
+ */
+
+namespace Drupal\commerce_product;
+
+use Drupal\commerce_product\Event\AvailableVariationsEvent;
+use Drupal\commerce_product\Event\ProductEvents;
+use Drupal\Core\Entity\EntityManagerInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class ProductVariationManager implements ProductVariationManagerInterface {
+
+  /**
+   * The variation storage.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected $variationStorage;
+
+  /**
+   * The event dispatcher.
+   *
+   * @var \Symfony\Component\EventDispatcher\EventDispatcherInterface
+   */
+  protected $eventDispatcher;
+
+  /**
+   * Constructs a ProductVariationManager object.
+   *
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entityManager
+   * @param \Symfony\Component\EventDispatcher\EventDispatcherInterface $eventDispatcher
+   */
+  public function __construct(EntityManagerInterface $entityManager, EventDispatcherInterface $eventDispatcher) {
+    $this->variationStorage = $entityManager->getStorage('commerce_product_variation');
+    $this->eventDispatcher = $eventDispatcher;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getEnabledVariations(ProductInterface $product) {
+    $variations = $product->variations;
```

Comment:
No need to declare $variations, you aren't going to use them again, just do foreach ($product->variations as $item) {